core/offchain/impl/offchain_worker_impl.hpp

```diff
@@ -0,0 +1,137 @@
+        std::shared_ptr<clock::SystemClock> clock,
+        std::shared_ptr<crypto::Hasher> hasher,
+        std::shared_ptr<storage::BufferStorage> storage,
+        std::shared_ptr<crypto::CSPRNG> random_generator,
+        std::shared_ptr<api::AuthorApi> author_api,
+        libp2p::Host &host,
+        std::shared_ptr<OffchainPersistentStorage> persistent_storage,
+        std::shared_ptr<runtime::Executor> executor,
+        const primitives::BlockHeader &header);
+
+    outcome::result<void> run() override;
+
+    bool isValidator() const override;
+
+    Result<Success, Failure> submitTransaction(
+        const primitives::Extrinsic &ext) override;
+
+    Result<OpaqueNetworkState, Failure> networkState() override;
+
+    Timestamp timestamp() override;
+
+    void sleepUntil(Timestamp timestamp) override;
+
+    RandomSeed randomSeed() override;
+
+    void localStorageSet(StorageType storage_type,
+                         const common::Buffer &key,
+                         common::Buffer value) override;
+
+    void localStorageClear(StorageType storage_type,
+                           const common::Buffer &key) override;
+
+    bool localStorageCompareAndSet(
+        StorageType storage_type,
+        const common::Buffer &key,
+        std::optional<std::reference_wrapper<const common::Buffer>> expected,
+        common::Buffer value) override;
+
+    outcome::result<common::Buffer> localStorageGet(
+        StorageType storage_type, const common::Buffer &key) override;
+
+    Result<RequestId, Failure> httpRequestStart(HttpMethod method,
+                                                std::string_view uri,
+                                                common::Buffer meta) override;
+
+    Result<Success, Failure> httpRequestAddHeader(
+        RequestId id, std::string_view name, std::string_view value) override;
+
+    Result<Success, HttpError> httpRequestWriteBody(
+        RequestId id,
+        common::Buffer chunk,
+        std::optional<Timestamp> deadline) override;
+
+    std::vector<HttpStatus> httpResponseWait(
+        const std::vector<RequestId> &ids,
+        std::optional<Timestamp> deadline) override;
+
+    std::vector<std::pair<std::string, std::string>> httpResponseHeaders(
+        RequestId id) override;
+
+    Result<uint32_t, HttpError> httpResponseReadBody(
+        RequestId id,
+        common::Buffer &chunk,
+        std::optional<Timestamp> deadline) override;
+
+    void setAuthorizedNodes(std::vector<libp2p::peer::PeerId> nodes,
+                            bool authorized_only) override;
+
+   private:
+    OffchainStorage &getStorage(StorageType storage_type);
+
+    const application::AppConfiguration &app_config_;
+    std::shared_ptr<clock::SystemClock> clock_;
+    std::shared_ptr<crypto::Hasher> hasher_;
+    std::shared_ptr<crypto::CSPRNG> random_generator_;
+    std::shared_ptr<api::AuthorApi> author_api_;
+    libp2p::Host &host_;
```

Comment:
As far as I can see host is only used to get peer info, maybe makes sense to replace it with `current_peer_info_` field?

Reply:
Good point. I'll change it.